File: 345-ReverseVowelsOfAString/345-ReverseVowelsOfAString.js
// Last updated: 3/7/2026, 10:02:37 PM
/**
 * @param {string} s
 * @return {string}
 */
var reverseVowels = function(s) {
     s = s.split('');
    let arr = [];
    let n; 

    for(let i=0; i<s.length; i++){
        if(s[i] == 'a' || s[i] == 'e' || s[i] == 'i' || s[i] == 'o' || s[i] == 'u' || s[i] == 'A' || s[i] == 'E' || s[i] == 'I' || s[i] == 'O' || s[i] == 'U'){
            arr.push(s[i]);
            
        }
    }

    n = arr.length;

    for(let j=0; j<s.length; j++){
        if(s[j] == 'a' || s[j] == 'e' || s[j] == 'i' || s[j] == 'o' || s[j] == 'u' || s[j] == 'A' || s[j] == 'E' || s[j] == 'I' || s[j] == 'O' || s[j] == 'U'){
            s[j] = arr[n-1];
            n--;
        }
    }
    s = s.join("")
    return s;
};

File: 345-ReverseVowelsOfAString/345-ReverseVowelsOfAString.cpp
// Last updated: 8/18/2026, 11:53:09 AM
class Solution {
public:
    string reverseVowels(string s) {
        string v = "";

        for(int i=0; i<s.size(); i++){
            if(s[i] == 'A' || s[i] == 'a' || s[i] == 'E' || s[i] == 'e' || s[i] == 'I' || s[i] == 'i' || s[i] == 'O' || s[i] == 'o' || s[i] == 'U' || s[i] == 'u'){
                v.push_back(s[i]);
            }
        }

        int n = v.size()-1;

        for(int i=0; i<s.size(); i++){
            if(s[i] == 'A' || s[i] == 'a' || s[i] == 'E' || s[i] == 'e' || s[i] == 'I' || s[i] == 'i' || s[i] == 'O' || s[i] == 'o' || s[i] == 'U' || s[i] == 'u'){
                s[i] = v[n];
                n--;
            }
        }
        return s;
    }
};